### PR TITLE
bmi270: add CAS and IOC

### PR DESCRIPTION
**Pull request by bkleiner**

### CODE REVIEW

File: src/drivers/drv_spi_bmi270.c [hunk 1]
```diff
@@ -145,4 +152,492 @@ void bmi270_read_data(uint8_t reg, uint8_t *data, uint32_t size) {
...
+  spi_seg_submit(&gyro_bus, NULL, gyro_segs);
+
+  const spi_txn_segment_t cas_segs[] = {
+      spi_make_seg_const(BMI270_REG_FEATURES_0_GYR_CAS | 0x80),
+      spi_make_seg_const(0xFF),
+      spi_make_seg_buffer(buf + 12, NULL, 2),
+  };
+  spi_seg_submit(&gyro_bus, NULL, cas_segs);
+
+  spi_txn_wait(&gyro_bus);
```
Comment (DusKing1) [inline src/drivers/drv_spi_bmi270.c:182]: I was wondering would this cas_segs contains two pull down on CS pin

File: src/drivers/drv_spi_bmi270.c [hunk 2]
```diff
@@ -145,4 +152,492 @@ void bmi270_read_data(uint8_t reg, uint8_t *data, uint32_t size) {
...
+      spi_make_seg_const(BMI270_REG_ACC_DATA_X_LSB | 0x80),
+      spi_make_seg_const(0xFF),
+      spi_make_seg_buffer(buf, NULL, 12),
+  };
+  spi_seg_submit(&gyro_bus, NULL, gyro_segs);
+
+  const spi_txn_segment_t cas_segs[] = {
+      spi_make_seg_const(BMI270_REG_FEATURES_0_GYR_CAS | 0x80),
+      spi_make_seg_const(0xFF),
+      spi_make_seg_buffer(buf + 12, NULL, 2),
```
Comment (DusKing1) [inline src/drivers/drv_spi_bmi270.c:178]: should be 4 bytes:
- Read byte: 1 byte
- Dummy byte: 1 byte
- Data bytes: 2 bytes, cuz you are reading a 16-bit register

File: src/drivers/drv_spi_bmi270.c (hunk 1 above)
Comment (bkleiner) [reply src/drivers/drv_spi_bmi270.c:182]: each `spi_seg_submit` here represents one spi transaction with one pull down.

File: src/drivers/drv_spi_bmi270.c (hunk 2 above)
Comment (bkleiner) [reply src/drivers/drv_spi_bmi270.c:178]: this is the case here. register and dummy, then 2 bytes read into a buffer.

File: src/drivers/drv_spi_bmi270.c (hunk 1 above)
Comment (DusKing1) [reply src/drivers/drv_spi_bmi270.c:182]: then no problem. you are reading the correct reg

File: src/drivers/drv_spi_bmi270.c (hunk 2 above)
Comment (DusKing1) [reply src/drivers/drv_spi_bmi270.c:178]: oh, misread. you are using a stack style method to build segment

**Review by DusKing1 — APPROVED** [review]
LGTM now

**Pull request merged**